providers/aws/services/vpc/vpc_endpoint_connections_trust_boundaries/vpc_endpoint_connections_trust_boundaries_test.py

```diff
@@ -0,0 +1,201 @@
+        ):
+            # Test Check
+            from providers.aws.services.vpc.vpc_endpoint_connections_trust_boundaries.vpc_endpoint_connections_trust_boundaries import (
+                vpc_endpoint_connections_trust_boundaries,
+            )
+
+            check = vpc_endpoint_connections_trust_boundaries()
+            result = check.execute()
+
+            assert result[0].status == "FAIL"
+            assert (
+                result[0].status_extended
+                == f"VPC Endpoint {vpc_endpoint['VpcEndpoint']['VpcEndpointId']} in VPC {vpc['VpcId']} has full access."
+            )
+            assert result[0].resource_id == vpc_endpoint["VpcEndpoint"]["VpcEndpointId"]
+
+    @mock_ec2
+    def test_vpc_endpoint_with_trusted_account(self):
+        # Create VPC Mocked Resources
+        ec2_client = client("ec2", region_name=AWS_REGION)
+
+        vpc = ec2_client.create_vpc(CidrBlock="10.0.0.0/16")["Vpc"]
+
+        route_table = ec2_client.create_route_table(VpcId=vpc["VpcId"])["RouteTable"]
+        vpc_endpoint = ec2_client.create_vpc_endpoint(
+            VpcId=vpc["VpcId"],
+            ServiceName="com.amazonaws.us-east-1.s3",
+            RouteTableIds=[route_table["RouteTableId"]],
+            VpcEndpointType="Gateway",
+            PolicyDocument=json.dumps(
+                {
+                    "Statement": [
+                        {
+                            "Effect": "Allow",
+                            "Principal": {"AWS": "arn:aws:iam::123456789012:root"},
+                            "Action": "*",
+                            "Resource": "*",
+                        }
+                    ]
+                }
+            ),
+        )
+        from providers.aws.lib.audit_info.audit_info import current_audit_info
+        from providers.aws.services.vpc.vpc_service import VPC
+
+        current_audit_info.audited_partition = "aws"
+        current_audit_info.audited_account = ACCOUNT_ID
+
+        with mock.patch(
+            "providers.aws.services.vpc.vpc_endpoint_connections_trust_boundaries.vpc_endpoint_connections_trust_boundaries.vpc_client",
+            new=VPC(current_audit_info),
+        ):
+            # Test Check
+            from providers.aws.services.vpc.vpc_endpoint_connections_trust_boundaries.vpc_endpoint_connections_trust_boundaries import (
+                vpc_endpoint_connections_trust_boundaries,
+            )
+
+            check = vpc_endpoint_connections_trust_boundaries()
+            result = check.execute()
+
+            assert result[0].status == "PASS"
+            assert (
+                result[0].status_extended
+                == f"Found trusted account {ACCOUNT_ID} in VPC Endpoint {vpc_endpoint['VpcEndpoint']['VpcEndpointId']} in VPC {vpc['VpcId']}."
+            )
+            assert result[0].resource_id == vpc_endpoint["VpcEndpoint"]["VpcEndpointId"]
+
+    @mock_ec2
+    def test_vpc_endpoint_with_trusted_account(self):
```

Comment:
This test has the same name than the one above

Reply:
What is the difference of both tests?

Reply:
Done.